### ISSUE TITLE
Example 12 in subset.R not working

### ISSUE THREAD

**Issue by j-harbin** [ArgoCanada/argoFloats#419]
There is a problem with subsetting by a range of cycles. See reprex below

``` r
library(argoFloats)
data(index)
A <- subset(index, cycle="124")
#> Kept 5 cycles (0.496%)
B <- subset(index, cycle=0:2)
#> Warning in grepl(sprintf("%03d", cycle), xcycle): argument 'pattern' has length
#> > 1 and only the first element will be used
#> Kept 1 cycles (0.0992%)
```

<sup>Created on 2021-03-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by dankelley** [comment]
J and I just fixed this in "develop" branch commit 94d117a68314c1516414082bf049b34d8b29e958 but we are keeping this open, so J can add a test to the test suite.  (That test can just just check the number.)

Also, J will check on whether it works on the "argos" type.

**Comment by j-harbin** [comment]
This was working fine for "argos" type, but I added a test in commit fda8ee17d477c49e9b0def33a6a2ae1de98a8716 of the develop branch.
